File: java_workplace/oce-adapter/src/main/java/com/xiaonei/talk/TalkRecentLoaderAdapter.java
package com.xiaonei.talk;

import java.util.Vector;

import com.xiaonei.talk.recent.loader.RecentLoaderManagerPrx;
import com.xiaonei.talk.recent.loader.RecentLoaderManagerPrxHelper;
import com.xiaonei.talk.recent.loader.TalkRecentUser;
import xce.util.channel.Channel;
import xce.util.service.ServiceAdapter;

public class TalkRecentLoaderAdapter extends ServiceAdapter {
	private static final String endpoints = "@TalkRecentLoader";
	private Vector<RecentLoaderManagerPrx> managersOneway = new Vector<RecentLoaderManagerPrx>();

	private static TalkRecentLoaderAdapter instance = new TalkRecentLoaderAdapter();

	public static TalkRecentLoaderAdapter instance() {
		return instance;
	}

	TalkRecentLoaderAdapter() {
		super(endpoints, 10, Channel.newTalkChannel(endpoints));

	}
	
	public RecentLoaderManagerPrx getManagerOneway(int id) {
		return locateProxy(managersOneway, "M", id, Channel.Invoke.Oneway,
				RecentLoaderManagerPrxHelper.class, 300);
	}
	
	public RecentLoaderManagerPrx getManager(int id) {
		return locateProxy(managersOneway, "M", id, Channel.Invoke.Twoway,
				RecentLoaderManagerPrxHelper.class, 300);
	}
	public void Update(int actor, TalkRecentUser groupinfo){
		RecentLoaderManagerPrx p = getManagerOneway(actor);
		p.Update(actor, groupinfo);
	}
	public void Delete(int actor, int id, int type){
		RecentLoaderManagerPrx p = getManagerOneway(actor);
		p.Delete(actor, id, type);
	}

}

File: main_project/OceCxxAdapter/src/TalkRecentLoaderAdapter.cpp
#include "TalkRecentLoaderAdapter.h"
#include "IceLogger.h"

using namespace com::xiaonei::talk;
using namespace com::xiaonei::talk::recent::loader;
using namespace MyUtil;


RecentLoaderManagerPrx TalkRecentLoaderAdapter::getManager(int id) {
	return getProxy(id);
}

RecentLoaderManagerPrx TalkRecentLoaderAdapter::getManagerOneway(int id) {
  return getProxyOneway(id);
}

void TalkRecentLoaderAdapter::Update(int actor, const TalkRecentUserPtr &groupinfo) {
	getManagerOneway(actor)->Update(actor, groupinfo);
}

TalkRecentUserSeq TalkRecentLoaderAdapter::LoadById(int owner) {
  return getManager(owner)->LoadById(owner);
}

TalkRecentUserSeqMap TalkRecentLoaderAdapter::LoadBySeq(const MyUtil::IntSeq &owners) {
	if (owners.empty()) {
		return TalkRecentUserSeqMap();
	}
	return getManager(owners.size())->LoadBySeq(owners);
}


